client/src/features/auth/authSlice.js

```diff
@@ -0,0 +1,87 @@
+import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
+import { login, register, getCurrentUser } from "./authAPI";
+
+const initialState = {
+  user: null,
+  isLoggedIn: false,
+  status: "idle",
+  error: null,
+};
+
+export const loginUser = createAsyncThunk("auth/login", async (credentials) => {
+  const response = await login(credentials);
+  localStorage.setItem("accessToken", response.data.accessToken);
+  const currentUserResponse = await getCurrentUser();
+  return currentUserResponse.data;
+});
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider error handling in login flow</b></div>
<div id="fix">

Consider handling potential network/API failures in the `loginUser` thunk. If the `login` call succeeds but `getCurrentUser` fails, we may end up with an inconsistent state where the token is set but user data is missing.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<blockquote>Check the AI-generated fix before applying</blockquote>
<div id="code">


```
 @@ -11,6 +11,15 @@
  async (credentials) => {
 -  const response = await login(credentials);
 -  localStorage.setItem("accessToken", response.data.accessToken);
 -  const currentUserResponse = await getCurrentUser();
 -  return currentUserResponse.data;
 +  try {
 +    const response = await login(credentials);
 +    localStorage.setItem("accessToken", response.data.accessToken);
 +    try {
 +      const currentUserResponse = await getCurrentUser();
 +      return currentUserResponse.data;
 +    } catch (error) {
 +      localStorage.removeItem("accessToken");
 +      throw error;
 +    }
 +  } catch (error) {
 +    throw error;
 +  }
  }
```

</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/Divyanshu9822/blog-mern-app/pull/22#issuecomment-2544587479>#245782</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged